src/machine/operation.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+/***
+  This file is part of systemd.
+
+  Copyright 2016 Lennart Poettering
+
+  systemd is free software; you can redistribute it and/or modify it
+  under the terms of the GNU Lesser General Public License as published by
+  the Free Software Foundation; either version 2.1 of the License, or
+  (at your option) any later version.
+
+  systemd is distributed in the hope that it will be useful, but
+  WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public License
+  along with systemd; If not, see <http://www.gnu.org/licenses/>.
+***/
+
+#include <sys/types.h>
+
+#include "sd-bus.h"
+#include "sd-event.h"
+
+#include "list.h"
+
+typedef struct Operation Operation;
```

Comment:
hmm, thinking about this more: I am pretty sure that older gcc versions complained if you defined the same type twice with "typedef", even if it had the identical definition both times. This is something that only ceased to be a problem on newer gcc versions. A quick google found me this for example:

https://gcc.gnu.org/ml/gcc-help/2006-02/msg00161.html

Now, if we place the typedef as you suggested we'd have to add it like this to every header file using it, right? This would make us a bit incompatible with those old gcc versions. No idea which ones were those though...